src/python/web/security.py

```
@@ -0,0 +1,242 @@
+
+def install_rate_limiting(app: bottle.Bottle, *, trust_x_forwarded_for: bool = False):
+    """
+    Per-IP sliding window rate limiter.  Returns 429 with Retry-After when
+    the limit is exceeded.  The SSE stream endpoint is exempt.
+
+    :param trust_x_forwarded_for: Only set True when deployed behind a trusted reverse proxy.
+    """
+    limiter = _RateLimiter()
+
+    @app.hook("before_request")
+    def _rate_limit():
```

Ruff (0.15.2)

[warning] Missing return type annotation for private function `_rate_limit`

Add return type annotation: `None`

(ANN202)